app/models/session.rb

```diff
@@ -125,10 +115,28 @@ def self.from_location(data)
     where(:streams => {:id => streams_ids.uniq})
   end
 
+  def self.filter_by_time_range(sessions, time_from, time_to)
+    time_from = Time.strptime(time_from.to_s, '%s')
+    time_to = Time.strptime(time_to.to_s, '%s')
```

Comment:
We should not allow url params to get this deep into the code unchecked. The controller should take all the params, validate them and convert to what deeper objects need. We should start moving in that direction asap.

This is an outdated example of that (https://github.com/3v0k4/rails_sane_architecture/blob/master/app/forms/comment_form.rb). The readme explains the structure https://github.com/3v0k4/rails_sane_architecture. I'd prolly use https://dry-rb.org/gems/dry-validation/ and https://dry-rb.org/gems/dry-struct/ nowadays for that